test/AppNotificationTests/BaseTestSuite.cpp

```diff
@@ -0,0 +1,602 @@
+{
+    if (Test::AppModel::IsPackagedProcess() || (!Test::AppModel::IsPackagedProcess() && !m_unregisteredFully))
+    {
+        VERIFY_IS_TRUE(EnsureNoActiveToasts());
+    }
+
+    if (!m_unregisteredFully)
+    {
+        UnregisterAllWithAppNotificationManager();
+    }
+
+    ::WindowsAppRuntime::VersionInfo::TestShutdown();
+    ::Test::Bootstrap::CleanupBootstrap();
+}
+
+void BaseTestSuite::RegisterWithAppNotificationManager()
+{
+    AppNotificationManager::Default().Register();
+    m_unregisteredFully = false;
+}
+
+void BaseTestSuite::UnregisterAllWithAppNotificationManager()
+{
+    AppNotificationManager::Default().UnregisterAll();
+    m_unregisteredFully = true;
+}
+
+void BaseTestSuite::VerifyRegisterActivatorandUnregisterActivator()
+{
+    RegisterWithAppNotificationManager();
+    AppNotificationManager::Default().Unregister();
+}
+
+void BaseTestSuite::VerifyFailedMultipleRegister()
+{
+    RegisterWithAppNotificationManager();
+    VERIFY_THROWS_HR(AppNotificationManager::Default().Register(), E_INVALIDARG);
+}
+
+void BaseTestSuite::VerifyUnregisterAll()
+{
+    RegisterWithAppNotificationManager();
+    UnregisterAllWithAppNotificationManager();
+}
+
+void BaseTestSuite::VerifyUnregisterTwice()
+{
+    RegisterWithAppNotificationManager();
+    AppNotificationManager::Default().Unregister();
+    VERIFY_THROWS_HR(AppNotificationManager::Default().Unregister(), E_NOT_SET);
+}
+void BaseTestSuite::VerifyToastSettingEnabled()
+{
+    VERIFY_ARE_EQUAL(AppNotificationManager::Default().Setting(), AppNotificationSetting::Enabled);
+}
+
+void BaseTestSuite::VerifyToastPayload()
+{
+    winrt::hstring xmlPayload{ L"<toast>intrepidToast</toast>" };
+    AppNotification toast{ xmlPayload };
+    VERIFY_ARE_EQUAL(toast.Payload(), xmlPayload);
+}
+
+void BaseTestSuite::VerifyToastTag()
+{
+    AppNotification toast{ CreateToastNotification() };
+    VERIFY_IS_TRUE(toast.Tag().empty());
+
+    toast.Tag(L"Tag");
+    VERIFY_ARE_EQUAL(toast.Tag(), L"Tag");
+}
+
+void BaseTestSuite::VerifyToastGroup()
+{
+    AppNotification toast{ CreateToastNotification() };
+    VERIFY_IS_TRUE(toast.Group().empty());
+
+    toast.Group(L"Group");
+    VERIFY_ARE_EQUAL(toast.Group(), L"Group");
+}
+
+void BaseTestSuite::VerifyToastProgressDataFromToast()
+{
+    AppNotification toast{ CreateToastNotification() };
+
+    AppNotificationProgressData progressData{ 1 };
+    progressData.Status(L"Status");
+    progressData.Title(L"Title");
+    progressData.Value(0.14);
+    progressData.ValueStringOverride(L"14%");
+    toast.Progress(progressData);
+
+    auto progressDataFromToast{ toast.Progress() };
+    VERIFY_ARE_EQUAL(progressDataFromToast.Status(), L"Status");
+    VERIFY_ARE_EQUAL(progressDataFromToast.Title(), L"Title");
+    VERIFY_ARE_EQUAL(progressDataFromToast.Value(), 0.14);
+    VERIFY_ARE_EQUAL(progressDataFromToast.ValueStringOverride(), L"14%");
+    VERIFY_ARE_EQUAL(progressDataFromToast.SequenceNumber(), (uint32_t) 1);
```

Comment:
As a general rule, always use modern C++ casts rather than C-styled.

Reply:
Curious why you cant pass the type directly here instead of casting.

Reply:
You can append the character u or U at the end of integer constant for unsigned int

Reply:
https://www.geeksforgeeks.org/integer-literal-in-c-cpp-prefixes-suffixes/

Reply:
Ah true that! Adding suffixes is actually better in this case 😊.